### PR TITLE
Symfony 4.0 support

### PR DESCRIPTION
**Pull request by cordoval**
ref https://docs.google.com/spreadsheets/d/1mFHQPp9uKtQTFjOBShIoPNHYgmkjHHdskdrd2C0BaF8/edit#gid=1730100865

### CODE REVIEW

**Comment by lsmith77** [comment]
aloha Luis!

could you also adjust travis-ci.yml accordingly?

File: .travis.yml [hunk 1]
```diff
@@ -29,12 +29,17 @@ matrix:
...
+      env: SYMFONY_VERSION=^3.3
+    - php: 7.1
+      env: SYMFONY_VERSION=^4.0
     - php: 5.4
       env: SYMFONY_VERSION=2.3;COMPOSER_FLAGS="--prefer-lowest"
 
 before_install:
   - if [[ $TRAVIS_PHP_VERSION != '7.0' && $TRAVIS_PHP_VERSION != '7.1' && $TRAVIS_PHP_VERSION != 'hhvm' ]]; then phpenv config-rm xdebug.ini; fi
   - if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then echo "memory_limit = -1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; fi
+  - if [ "$SYMFONY_VERSION" = "^4.0" ]; then composer config minimum-stability dev; composer config prefer-stable true; fi
```
Comment (lsmith77) [inline .travis.yml]: I think we can actually set those two settings in composer.json in general

Comment (iltar) [reply .travis.yml]: I would remove the `prefer-stable`. I've noticed that this prevents 4.0 from being installed as dev/beta and you can't unset this setting once in the json.

Comment (cordoval) [reply .travis.yml]: interesting, this seems to work elsewhere

Comment (lsmith77) [reply .travis.yml]:
```
diff --git a/.travis.yml b/.travis.yml
index 7c226eb..9af5bcb 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -39,7 +39,6 @@ matrix:
 before_install:
   - if [[ $TRAVIS_PHP_VERSION != '7.0' && $TRAVIS_PHP_VERSION != '7.1' && $TRAVIS_PHP_VERSION != 'hhvm' ]]; then phpenv config-rm xdebug.ini; fi
   - if [ "$TRAVIS_PHP_VERSION" != "hhvm" ]; then echo "memory_limit = -1" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini; fi
-  - if [ "$SYMFONY_VERSION" = "^4.0" ]; then composer config minimum-stability dev; composer config prefer-stable true; fi
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require --no-update symfony/symfony=$SYMFONY_VERSION; fi
 
 install: composer update --prefer-source $COMPOSER_FLAGS
diff --git a/composer.json b/composer.json
index 1b2357f..af49610 100644
--- a/composer.json
+++ b/composer.json
@@ -22,6 +22,8 @@
             "homepage": "https://github.com/liip/LiipMonitorBundle/contributors"
         }
     ],
+    "minimum-stability": "dev",
+    "prefer-stable": true,
     "require": {
         "php": "^5.4|^7.0",
         "symfony/framework-bundle": "~2.3|~3.0|^4.0",
```

as for the failures .. I think we need a new release of `matthiasnoback/symfony-dependency-injection-test` that allows Symfony 4.0

**Comment by cordoval** [comment]
@lsmith77 you are going to have to forgive me, i cannot pull time for this now even though i also do need it to be updated. I gotta deliver here. Sorry for passing out this time

**Comment by lsmith77** [comment]
no worries .. follow up in https://github.com/liip/LiipMonitorBundle/pull/153

**Pull request closed without merging**